src/main/scala/com/fulcrumgenomics/vcf/api/VcfSource.scala

```diff
@@ -42,9 +42,10 @@ import htsjdk.variant.vcf.{VCFCodec, VCFFileReader, VCFHeader}
   *
   * @param reader the underlying HTSJDK [[VCFFileReader]]
   */
-class VcfSource private(private val reader: AbstractFeatureReader[VariantContext, _]) extends View[Variant] with Closeable {
+class VcfSource private(private val reader: AbstractFeatureReader[VariantContext, _],
+                        _header: Option[VcfHeader]) extends View[Variant] with Closeable {
```

Comment:
I get why you need something like this but I'm not a big fan of it.  I'd like to suggest something a bit different:

1. How about we have it take an optional `headerTransformer: VcfHeader => VcfHeader` instead of a replacement header?  I think they're functionally equivalent, but this makes it clear the purpose is to allow you to fix/adjust the header rather than implying things might still work with a wholly different header.
2. We only expose it in the constructor and not the object.apply() so it's semi-hidden, or
3. We put it in _an_ apply but have two separate apply methods so it's super-obvious that you don't need it?